docs/bundler/executables.md

```diff
@@ -45,6 +45,58 @@ When deploying to production, we recommend the following:
 bun build --compile --minify --sourcemap ./path/to/my/app.ts --outfile myapp
 ```
 
+### Cross-compile to other platforms
+
+To target a different platform, architecture, or version of Bun, use the `--target` flag.
+
+To build for Linux from macOS, Windows or Linux, you can use the following command:
+
+```sh
+# The default architecture is x64 if an os is specified.
+bun build --compile --target=bun-linux ./index.ts --outfile myapp
+
+# Optionally specify an architecture:
+bun build --compile --target=bun-linux-x64 ./index.ts --outfile myapp
+bun build --compile --target=bun-linux-arm64 ./index.ts --outfile myapp
+
+# Optionally specify a version of Bun to use:
+bun build --compile --target=bun-linux-x64-v1.1.5 ./index.ts --outfile myapp
+bun build --compile --target=bun-linux-arm64-v1.1.5 ./index.ts --outfile myapp
+```
+
+To build for Windows x64:
+
+```sh
+bun build --compile --target=bun-windows-x64 ./path/to/my/app.ts --outfile myapp
+```
+
+To build for macOS arm64:
+
+```sh
+bun build --compile --target=bun-darwin-arm64 ./path/to/my/app.ts --outfile myapp
+```
+
+To build for macOS x64:
+
+```sh
+bun build --compile --target=bun-darwin-x64 ./path/to/my/app.ts --outfile myapp
+```
+
+#### Supported targets
+
+The order of the `--target` flag does not matter, as long as they're delimited by a `-`.
+
+| --target              | Operating System | Architecture | Modern | Baseline |
+| --------------------- | ---------------- | ------------ | ------ | -------- |
+| bun-linux-x64         | Linux            | x64          | ✅     | ✅       |
+| bun-linux-arm64       | Linux            | arm64        | ✅     | N/A      |
+| bun-windows-x64       | Windows          | x64          | ✅     | ✅       |
+| ~~bun-windows-arm64~~ | Windows          | arm64        | ❌     | ❌       |
+| bun-darwin-x64        | macOS            | x64          | ✅     | ✅       |
+| bun-darwin-arm64      | macOS            | arm64        | ✅     | N/A      |
+
+On x64 platforms, Bun uses SIMD optimizations which require a modern CPU supporting AVX2 instructions. The `-baseline` build of Bun is for older CPUs that don't support these optimizations. Normally, when you install Bun we automatically detect which version to use but this can be harder to do when cross-compiling since you might not know the target CPU. You usually don't need to worry about it on Darwin x64, but it is relevant for Windows x64 and Linux x64. If you or your users see `"Illegal instruction"` errors, you might need to use the baseline version.
```

Comment:
We should not make this implicit based on the machine that the command is running on. Better to add a `--baseline` flag. 

What I want to avoid here is running the same command but getting different binaries based on your computer, we want this to be reproducible.